WordPress/gutenberg
# Add site logo crop

## Description


Part of #30406.

Also fixes #29217 in 93dc9675cb448586621a840a310202fe3acdee65. This can be moved to another PR if people would prefer to keep the fix separate.

Adds cropping to the site logo by:

1. Moving the cropping component to the @wordpress/block-editor package and exporting it as an `__experimental` component
2. Using the exported cropper in the SiteLogo block

## How has this been tested?




1. Add SiteLogo block.
2. See that toolbar doesn't appear when the logo is empty.
3. Apply a crop when a logo is present.
4. See that the site logo is updated when the crop is applied.

## Screenshots 

<img width="210" alt="image" src="https://user-images.githubusercontent.com/5129775/117477980-55d4d280-aefa-11eb-98df-44fa61dd49ba.png">

<img width="310" alt="image" src="https://user-images.githubusercontent.com/5129775/117478010-5cfbe080-aefa-11eb-8668-3006a02154e5.png">

![crop-logo](https://user-images.githubusercontent.com/5129775/117478490-eca18f00-aefa-11eb-9d1e-9337017a8c53.gif)

## Types of changes





New feature

## Checklist:
